### PR TITLE
A proposal to alter the builder order

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
## Summary

A proposal to alter the builder order and move Node.js lower.

## Use Cases

This prevents Node.js buildpack from incorrectly detecting true for applications that also bundle frontend code with backend code.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by fg-j** [comment]
Personally, I'd like to do a bit more testing of the updated order before merging this in – in particular, run against more sample apps and test a reordering of the full builder, not just the base builder.

But on a meta level, I also wonder: how much are users relying on the builders rather than defining their own? Which types of users are relying on the builder? The answer to these questions will influence how important it is to support the frontend-backend combo use-case described here *and* how much we should actually worry about breaking other users' builds.

**Comment by fg-j** [comment]
@dmikusa-pivotal This sounds reasonable to me. In addition, it seems to me that most Paketo users, right now, are using CNB platforms other than `kpack`, which has its own builder abstraction. 

I'd still like to see more validation that the proposed reordering is minimally breaking for the types of apps we already support.

File: text/0000-adjust-builder-order.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    version = "5.0.2"
+```
+
+This will enable other buildpacks to detect before Node.js, that way if you have frontend and backend code in the same application then the correct buildpack will be selected.
+
+The chagne to the order will not impact applications that only build using a single language family.
```
Comment (ekcasey) [inline text/0000-adjust-builder-order.md:43]:
```suggestion
The change to the order will not impact applications that only build using a single language family.
```

**Review by ekcasey — APPROVED** [review]

File: text/0000-adjust-builder-order.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Prior Art
+
+The [Dotnet Core buildpack was placed earlier in the detction list for similar reasons](https://github.com/paketo-buildpacks/rfcs/blob/009ff161f7b4f05c354766eb89e0a218eef5e929/text/builders/0001-buildpack-order.md).
+
+## Unresolved Questions and Bikeshedding
+
+1. How can we be sure that this won't create other problems?
+
+    - A [test was done](https://github.com/paketo-buildpacks/base-builder/issues/341#issuecomment-1004414372) with the new order and Paketo Samples all still build OK.
+    - Do we need to test further?
```
Comment (fg-j) [inline text/0000-adjust-builder-order.md:66]: I'd like to try testing with the Paketo Full builder + all of the sample apps it supports before I approve this.

Comment (dmikusa-pivotal) [reply text/0000-adjust-builder-order.md:66]: I ran through the sample test suites with the Paketo Full Builder, where I moved the Node.js composite entry to almost the end, just before Procfile. All the tests passed, with the exception of one.

```
Passed: 5 | Failed: 1 | Skipped: 0

--- FAIL: TestNodejs (112.53s)
    --- FAIL: TestNodejs/Nodejs (0.00s)
        --- PASS: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/no_package_manager/builds_successfully (12.70s)
        --- PASS: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/app_uses_yarn/builds_successfully (19.68s)
        --- PASS: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/app_uses_npm/builds_successfully (15.59s)
        --- PASS: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/app_uses_vue_and_npm/builds_successfully (85.26s)
        --- FAIL: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/app_uses_angular_and_npm/builds_successfully (97.46s)
        --- PASS: TestNodejs/Nodejs/Nodejs_with_paketobuildpacks/builder:full_builder/detects_a_Node.js_app/app_uses_react_and_yarn/builds_successfully (99.83s)
FAIL
FAIL	github.com/paketo-buildpacks/samples/nodejs	112.681s
FAIL
```

It failed saying:

```
../layers/paketo-buildpacks_npm-install/modules/node_modules/css-loader/dist/runtime/api.js - Error: Module build failed (from ../layers/paketo-buildpacks_npm-install/modules/node_modules/@ngtools/webpack/src/ivy/index.js):
                    Transform failed with 1 error:
                    error: Invalid version: "15.2-15.3"
```

which seems unrelated to the changes I made. It also fails in the same way when I run using an unmodified full builder.

**Comment by fg-j** [comment]
If we're going to re-order the base and full builders, should we cut minor releases upon this change?

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**